contracts/mainnet/connectors/uniswapV3/helpers.sol

```diff
@@ -0,0 +1,229 @@
+        uint256 amtB;
+        uint256 slippage;
+    }
+
+    /**
+     * @dev Get Last NFT Index
+     * @param user: User address
+     */
+    function _getLastNftId(address user)
+        internal
+        view
+        returns (uint256 tokenId)
+    {
+        uint256 len = nftManager.balanceOf(user);
+        tokenId = nftManager.tokenOfOwnerByIndex(user, len - 1);
+        console.log("tokenId", tokenId);
+    }
+
+    function getMinAmount(
+        TokenInterface token,
+        uint256 amt,
+        uint256 slippage
+    ) internal view returns (uint256 minAmt) {
+        uint256 _amt18 = convertTo18(token.decimals(), amt);
+        minAmt = wmul(_amt18, sub(WAD, slippage));
+        minAmt = convert18ToDec(token.decimals(), minAmt);
+    }
+
+    /**
+     * @dev Mint function which interact with Uniswap v3
+     */
+    function _mint(MintParams memory params)
+        internal
+        returns (
+            uint256 tokenId,
+            uint128 liquidity,
+            uint256 amount0,
+            uint256 amount1
+        )
+    {
+        (TokenInterface _token0, TokenInterface _token1) = changeEthAddress(
+            params.tokenA,
+            params.tokenB
+        );
+
+        uint256 _amount0 = params.amtA == uint256(-1)
+            ? getTokenBal(TokenInterface(params.tokenA))
+            : params.amtA;
+        uint256 _amount1 = params.amtB == uint256(-1)
+            ? getTokenBal(TokenInterface(params.tokenB))
+            : params.amtB;
+
+        uint256 isEth = address(_token0) == wethAddr ? 0 : 2;
+        isEth = address(_token1) == wethAddr ? 1 : 2;
+        convertEthToWeth(isEth == 0, _token0, _amount0);
+        convertEthToWeth(isEth == 1, _token1, _amount1);
+
+        approve(_token0, address(nftManager), _amount0);
+        approve(_token1, address(nftManager), _amount1);
+
+        uint256 _minAmt0 = getMinAmount(_token0, _amount0, params.slippage);
+        uint256 _minAmt1 = getMinAmount(_token1, _amount1, params.slippage);
+
+        INonfungiblePositionManager.MintParams
+            memory params = INonfungiblePositionManager.MintParams(
+                address(_token0),
+                address(_token1),
+                params.fee,
+                params.tickLower,
+                params.tickUpper,
+                _amount0,
+                _amount1,
+                _minAmt0,
+                _minAmt1,
+                address(this),
+                block.timestamp
+            );
+
+        (tokenId, liquidity, amount0, amount1) = nftManager.mint(params);
+    }
+
+    /**
+     * @dev Check if token address is etherAddr and convert it to weth
+     */
+    function _checkETH(
+        uint256 _tokenId,
+        uint256 _amount0,
+        uint256 _amount1
+    ) internal {
+        (, , address token0, address token1, , , , , , , , ) = nftManager
+            .positions(_tokenId);
+        uint256 isEth = token0 == wethAddr ? 0 : 2;
+        isEth = token1 == wethAddr ? 1 : 2;
```

Comment:
```suggestion
        uint256 isEth = token0 == wethAddr ? 0 : 2;
        isEth = token1 == wethAddr ? 1 : 2;
```

Assume: token0 is weth and token1 is dai. This logic won't work. As line:123 overide line:122 logic.

Reply:
Okay, got it.